components/faction/commons/faction.lua

```diff
@@ -74,13 +74,14 @@ end
 --- Parses multiple factions from input.
 -- Returns an array of faction identifiers.
 -- Returns an empty array for nil input. Throws upon invalid inputs.
----@param input string
+---@param input string?
 ---@param options {sep: string?, alias: boolean?}?
 ---@return table
 function Faction.readMultiFaction(input, options)
 	if String.isEmpty(input) then
 		return {}
 	end
+	---@cast input -nil
```

Comment:
What would happen without the string is empty`check?

Reply:
the `mw.text.split` some lines below would error